### PR TITLE
fix: breaking ethpm types

### PR DESCRIPTION
**Pull request by fubuloubu**
### What I did
Breaking changes introduced in `ethpm-types` b4/b5. This fixes them.

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: src/ape/contracts/base.py [hunk 1]
```diff
@@ -284,43 +284,61 @@ def __getattr__(self, attr_name: str) -> Any:
...
+                has_matching_events,
             ]
+        )
+
+        if num_matching_conditions == 0:
+            # Didn't find anything that matches
+            # NOTE: `__getattr__` *must* raise `AttributeError`
+            raise AttributeError(f"{self.__class__.__name__} has no attribute '{attr_name}'.")
+
+        if num_matching_conditions > 1:
```
Comment (unparalleled-js) [inline src/ape/contracts/base.py:312]: doesn't really matter but could do `elif` here. Increase readability a tad.

Comment (fubuloubu) [reply src/ape/contracts/base.py:312]:
```suggestion
        elif num_matching_conditions > 1:
```

File: src/ape/contracts/base.py [hunk 2]
```diff
@@ -284,43 +284,61 @@ def __getattr__(self, attr_name: str) -> Any:
...
+        def name_matches(abi):
+            return abi.name == attr_name
+
+        selected_view_methods = list(filter(name_matches, self._contract_type.view_methods))
+        has_matching_view_methods = len(selected_view_methods) > 0
+
+        selected_mutable_methods = list(filter(name_matches, self._contract_type.mutable_methods))
+        has_matching_mutable_methods = len(selected_mutable_methods) > 0
+
+        selected_events = list(filter(name_matches, self._contract_type.events))
```
Comment (unparalleled-js) [inline src/ape/contracts/base.py:296]: should we raise an error if for some reason we match multiple sets here?

**Review by unparalleled-js — COMMENTED** [review]
Looks guuuud

**Review by evanblank3 — APPROVED** [review]
[unbreak my heart](https://www.youtube.com/watch?v=p2Rch6WvPJE)

**Comment by fubuloubu** [comment]
breaking `ape-solidity`:

```
userdoc
  value is not a valid dict (type=type_error.dict)
devdoc
  value is not a valid dict (type=type_error.dict)
```

**Pull request merged**